### PR TITLE
Random test migrations to Protobuf

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Brings tests in line with Android.

### CODE REVIEW

File: Firestore/core/test/unit/remote/datastore_test.cc [hunk 1]
```diff
@@ -216,10 +216,10 @@ TEST_F(DatastoreTest, CommitMutationsSuccess) {
 
 TEST_F(DatastoreTest, LookupDocumentsOneSuccessfulRead) {
   bool done = false;
-  std::vector<MaybeDocument> resulting_docs;
+  std::vector<Document> resulting_docs;
   Status resulting_status;
   datastore->LookupDocuments(
-      {}, [&](const StatusOr<std::vector<MaybeDocument>>& maybe_documents) {
+      {}, [&](const StatusOr<std::vector<Document>>& maybe_documents) {
```
Comment (thebrianchen) [inline Firestore/core/test/unit/remote/datastore_test.cc:222]: s/maybe_documents/documents

File: Firestore/core/test/unit/remote/datastore_test.cc [hunk 2]
```diff
@@ -237,16 +237,16 @@ TEST_F(DatastoreTest, LookupDocumentsOneSuccessfulRead) {
...
 }
 
 TEST_F(DatastoreTest, LookupDocumentsTwoSuccessfulReads) {
   bool done = false;
-  std::vector<MaybeDocument> resulting_docs;
+  std::vector<Document> resulting_docs;
   Status resulting_status;
   datastore->LookupDocuments(
-      {}, [&](const StatusOr<std::vector<MaybeDocument>>& maybe_documents) {
+      {}, [&](const StatusOr<std::vector<Document>>& maybe_documents) {
```
Comment (thebrianchen) [inline Firestore/core/test/unit/remote/datastore_test.cc:249]: same here and below

File: Firestore/core/test/unit/model/mutation_test.cc [hunk 3]
```diff
@@ -51,110 +50,112 @@ using testutil::WrapObject;
...
-  auto result = transform.ApplyToLocalView(base_doc, now);
+  transform.ApplyToLocalView(doc, now);
 
   // Server timestamps aren't parsed, so we manually insert it.
   ObjectValue expected_data =
       WrapObject("foo", Map("bar", "<server-timestamp>"), "baz", "baz-value");
-  expected_data =
-      expected_data.Set(Field("foo.bar"), FieldValue::FromServerTimestamp(now));
+  expected_data.Set(Field("foo.bar"),
+                    EncodeServerTimestamp(now, absl::nullopt));
```
Comment (thebrianchen) [inline Firestore/core/test/unit/model/mutation_test.cc:151]: Not sure if you have another PR to fix this, but `now` has type `Timestamp`, but `EncodeServerTimestamp` requires `google_protobuf_Timestamp`.

**Review by thebrianchen — APPROVED** [review]
lgtm

File: Firestore/core/test/unit/model/mutation_test.cc (hunk 3 above)
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/model/mutation_test.cc:151]: Yes - https://github.com/firebase/firebase-ios-sdk/pull/7991/files#diff-8ca9b7524451f1805e18521af3ffeec7b82dc053437f2f55c4d127259eaa2e5eL32

File: Firestore/core/test/unit/remote/datastore_test.cc (hunk 2 above)
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/remote/datastore_test.cc]: Fixed

File: Firestore/core/test/unit/remote/datastore_test.cc (hunk 1 above)
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/remote/datastore_test.cc]: Fixed

**Pull request merged**